org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/ScalaLabelProvider.scala

```diff
@@ -0,0 +1,114 @@
+package org.scalaide.ui.internal.editor
+
+/**
+ * @author ailinykh
+
+
+ */
+import org.eclipse.jface.viewers.ILabelProvider
+import java.io.FileInputStream
+import org.eclipse.swt.graphics.Image
+import org.eclipse.jface.viewers.ILabelProviderListener;
+import org.scalaide.core.internal.ScalaPlugin
+
+class ScalaLabelProvider extends ILabelProvider{
+
+
+    /**
+   * Gets the image to display for a node in the tree
+   *
+   * @param arg0
+   *            the node
+   * @return Image
+   */
+  override def  getImage( o:Object):Image= {
+    import org.scalaide.ui.ScalaImages._
+
+    val reg = ScalaPlugin().imageDescriptorRegistry
+    o match {
+      case n:ClassNode => if(n.isTrait) reg.get( SCALA_TRAIT) else reg.get( SCALA_CLASS)
+      case n:ValNode => if(n.isPrivate) reg.get( PRIVATE_VAL) else if(n.isProtected) reg.get(PROTECTED_VAL) else reg.get(PUBLIC_VAL)
+      case n:VarNode => if(n.isPrivate) reg.get( PRIVATE_VAR) else if(n.isProtected) reg.get(PROTECTED_VAR) else reg.get(PUBLIC_VAR)
+      case n:MethodNode => if(n.isPrivate) reg.get( PRIVATE_DEF) else if(n.isProtected) reg.get(PROTECTED_DEF) else reg.get(PUBLIC_DEF)
+      case n:ObjectNode => reg.get(SCALA_OBJECT)
+      case n:PackageNode => reg.get(PACKAGE)
+      case _ => null
+    }
+  }
+
+  /**
+   * Gets the text to display for a node in the tree
+   *
+   * @param arg0
+   *            the node
+   * @return String
+   */
+  override def  getText( o:Object):String ={
```

Comment:
I think most of the implementation of this method is needless. We already have `DeclarationPrinter`. It should be used here too.